src/L2/paused/L2GovernorPaused.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.23;
+
+import { Ownable2StepUpgradeable } from "@openzeppelin-upgradeable/contracts/access/Ownable2StepUpgradeable.sol";
+import { Initializable } from "@openzeppelin-upgradeable/contracts/proxy/utils/Initializable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import { TimelockControllerUpgradeable } from
+    "@openzeppelin-upgradeable/contracts/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+import { L2Governor } from "../L2Governor.sol";
+
+contract L2GovernorPaused is L2Governor {
+    /// @notice Setting global params.
+    function initializePaused() public reinitializer(2) { }
+
+    /// @notice Marking version as paused.
+    function version() public pure virtual override returns (string memory) {
+        return "1.0.0-paused";
+    }
+
+    /// @notice Override the cancel function to pause Governor interactions.
+    function cancel(
+        address[] memory,
+        uint256[] memory,
+        bytes[] memory,
+        bytes32
+    )
+        public
+        virtual
+        override
+        returns (uint256)
+    {
+        revert("L2GovernorPaused: Governor is paused");
+    }
+
+    /// @notice Override the castVote function to pause Governor interactions.
+    function castVote(uint256, uint8) public virtual override returns (uint256) {
+        revert("L2GovernorPaused: Governor is paused");
```

Comment:
Using custom errors is [a bit cheaper](https://soliditylang.org/blog/2021/04/21/custom-errors/) than using `revert` with a string. This is maybe not so important for our L2. But it seems to be a better practice.
Feel free to ignore if you disagree.

Reply:
I would leave it as it is.

Reply:
Although I also found Custom Errors more elegant, I don't have strong opinion here.

Reply:
As you've already changed it to custom error, I'm fine with this change.